standard/lpdb/condition.lua

```diff
@@ -15,18 +15,20 @@ local Condition = {}
 -- Abstract class, node of the conditions tree
 local _ConditionNode = Class.new()
 
---[[
-	A tree of conditions, specifying the conditions for an LPDB request
-
-	Can be used recursively, as in, a tree of trees
-]]
+---A tree of conditions, specifying the conditions for an LPDB request.
+---Can be used recursively, as in, a tree of trees.
+---@class ConditionTree
+---@field _nodes ConditionNode[]
+---@field booleanOperator 'AND'|'OR'
 local ConditionTree = Class.new(_ConditionNode,
 	function(self, booleanOperator)
 		self.booleanOperator = booleanOperator
 		self._nodes = {}
 	end
 )
 
+---@param node ConditionNode|ConditionNode[]
```

Comment:
```suggestion
---@param node _ConditionNode|ConditionNode[]
```
It can take a Tree in addition to a Node, so use the abtract class

Reply:
causes anno warning about undefined field `is_a`, so need to add that too